src/App.js

```diff
@@ -1,41 +1,96 @@
-import {useEffect} from "react";
-import './App.css';
-import { RestApiCalls } from './RestApiCalls';
-import { useVideoList } from "./Context/VideoLibraryContext";
-import { VideoListing } from "./VideoListing/VideoListing";
-import { TopNavBar } from "./Navbar/TopNavBar.jsx";
-import { SideBar } from "./Navbar/SideBar";
-import {Switch,Route} from "react-router-dom";
-import { useMyPlaylist } from "./Context/MyPlaylistContext";
-
-
-function App() {
-  const {videoLibrary,videoLibraryDispatch} = useVideoList();
-  const {myPlaylist,myPlaylistDispatch} = useMyPlaylist();
-
-  useEffect(()=>{
-    (async function(){
-      const response = await RestApiCalls("GET","/api/videos")
-      videoLibraryDispatch({type:"LOAD_VIDEOLIST",payload:response.data})
-    })()
-
-    console.log({videoLibrary});
-
-  },[])
-  console.log({myPlaylist})
-  return (
-    <div className={`App`}>
-      {videoLibrary.showModal && <div className = "background-overlay"></div>}
-      <TopNavBar/>
-      <SideBar/>
-      {videoLibrary.videoList.length === 0 && <p>Data not loaded yet</p>}
-      <Switch>
-        <Route path = "/" component = {VideoListing}/>
-      </Switch>
-      
-      
-    </div>
-  );
-}
-
-export default App;
+import { useEffect, useState } from 'react'
+import './App.css'
+import { RestApiCalls } from './utils/callRestApi'
+import { useVideoList } from './Context/VideoLibraryContext'
+import { VideoListing } from './VideoListing/VideoListing'
+import { TopNavBar } from './Navbar/TopNavBar.jsx'
+import { Routes, Route } from 'react-router-dom'
+import { useMyPlaylist } from './Context/MyPlaylistContext'
+import { WatchLater } from './Watch Later/WatchLater'
+import { Library } from './Library/Library'
+import { StackedListTemplate } from './StackedListTemplate/StackedListTemplate'
+import { PlayVideoPage } from './PlayVideoPage/PlayVideoPage'
+import { BACKEND } from './utils/api'
+import {Login} from './Login/Login';
+import {SignUp} from './Login/SignUp';
+import { useAuth } from './Context/AuthContext';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import axios from 'axios'
+import { PrivateRoutes } from './Navbar/PrivateRoutes'
+
+	function App() {
+		const { videoLibrary, videoLibraryDispatch } = useVideoList()
+		const {  myPlaylistDispatch } = useMyPlaylist()
+		const [loading, setLoading] = useState(true)
+		const { auth } = useAuth();
+
+
+		const fetchPlaylist = async () => {
+
+			const { response } = await RestApiCalls('GET', `${BACKEND}/${auth.user._id}/playlists`)
```

Comment:
This is super useful, putting URLs in a different file, and also creating a function for handling server calls!